core/services/fetchers.go

```diff
@@ -0,0 +1,168 @@
+package services
+
+import (
+	"chainlink/core/logger"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/guregu/null"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+)
+
+// Fetcher is the interface encapsulating all functionality needed to retrieve
+// a price.
+type Fetcher interface {
+	Fetch() (float64, error)
+}
+
+// httpFetcher retrieves data via HTTP from an external price adapter source.
+type httpFetcher struct {
+	client      *http.Client
+	url         *url.URL
+	requestData string
+}
+
+func newHTTPFetcher(
```

Comment:
Maybe rename to price fetcher or something similar?  Now the entity is moved into a larger domain more specific language may help

Reply:
Also, what about changing the file name to fluxmonitor_fetchers.go?  since this code is here to support the flux monitor.

Reply:
I'm conflicted on this suggestion, which means I'm leaning towards leaving it as is.

Following the go interface naming convention (https://golang.org/doc/effective_go.html#interface-names):
> By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.

I end up with `Fetcher`. HTTPFetcher is merely the HTTP version of the interface following said naming convention. Injecting a `Price` in there seems to break the pattern.